cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -278,15 +319,19 @@ public static void setDynamicPartitionerConcurrency(Map<String, String> argument
    * @return whether concurrent writers are allowed in the case of DynamicPartitioner.
    */
   public static boolean isDynamicPartitionerConcurrencyAllowed(Map<String, String> arguments) {
-    if (getDynamicPartitioner(arguments) == null) {
-      throw new IllegalArgumentException("Cannot retrieve a setting of DynamicPartitioner, without first setting " +
-                                           "a DynamicPartitioner.");
-    }
+    ensureDynamicPartitionerConfigured(arguments);
     String concurrencyAllowed = arguments.get(DYNAMIC_PARTITIONER_ALLOW_CONCURRENCY);
     if (concurrencyAllowed == null) {
     // default to true
       return true;
     }
     return Boolean.valueOf(concurrencyAllowed);
   }
+
+  private static void ensureDynamicPartitionerConfigured(Map<String, String> args) {
+    if (getDynamicPartitioner(args) == null) {
+      throw new IllegalArgumentException("Cannot retrieve a setting of DynamicPartitioner, without first setting " +
```

Comment:
this is used both for setting and retrieving. In the first case, this message would be misleading